features/steps/jobs_steps.go

```diff
@@ -214,3 +246,103 @@ func (f *JobsFeature) iCallGETJobsidUsingTheGeneratedId() error {
 
 	return f.ErrorFeature.StepError()
 }
+
+//iHaveGeneratedThreeJobsInTheJobStore is a feature step that can be defined for a specific JobsFeature.
+//It calls POST /jobs with an empty body, three times, which causes three default job resources to be generated.
+func (f *JobsFeature) iHaveGeneratedThreeJobsInTheJobStore() error {
+	//call POST /jobs three times
+	f.callPostJobs()
+	f.callPostJobs()
+	f.callPostJobs()
+
+	return f.ErrorFeature.StepError()
+}
+
+//iWouldExpectThereToBeThreeOrMoreJobsReturnedInAList is a feature step that can be defined for a specific JobsFeature.
+//It checks the response from calling GET /jobs to make sure that a list containing three or more jobs has been returned.
+func (f *JobsFeature) iWouldExpectThereToBeThreeOrMoreJobsReturnedInAList() error {
+	f.responseBody, _ = ioutil.ReadAll(f.ApiFeature.HttpResponse.Body)
+
+	var response models.Jobs
+	err := json.Unmarshal(f.responseBody, &response)
+	if err != nil {
+		return err
+	}
+	numJobsFound := len(response.Job_List)
+	assert.True(&f.ErrorFeature, numJobsFound >= 3, "The list should contain three or more jobs but it only contains "+strconv.Itoa(numJobsFound))
+
+	return f.ErrorFeature.StepError()
+}
+
+//inEachJobIWouldExpectIdLast_updatedAndLinksToHaveThisStructure is a feature step that can be defined for a specific JobsFeature.
+//It checks the response from calling GET /jobs to make sure that each job contains the expected types of values of id,
+//last_updated, and links.
+func (f *JobsFeature) inEachJobIWouldExpectIdLast_updatedAndLinksToHaveThisStructure(table *godog.Table) error {
+	assist := assistdog.NewDefault()
+	expectedResult, err := assist.ParseMap(table)
+	if err != nil {
+		panic(err)
+	}
+	var response models.Jobs
+
+	err = json.Unmarshal(f.responseBody, &response)
+	if err != nil {
+		return err
+	}
+
+	for j := range response.Job_List {
+		job := response.Job_List[j]
+		err2 := f.checkStructure(err, job.ID, job.LastUpdated, expectedResult, job.Links)
```

Comment:
I don't see it being necessary to create a new error type variable assigned to checkStructure method. We could just use existing error, err?

Reply:
yes I agree - changed now